From bc98a10b4ce45881c98c1297b2a56a786ea09b54 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Mon, 18 Feb 2019 14:00:11 +0800
Subject: [PATCH] DRVLIB_SW-1485 - Switch 10/100M to use default LMAC instead
 of XGMAC

Xgmac supports only XGMII and GMII, while LMAC supports GMII
and MII.
Setting all speed mode to XGMAC caused problem when we connect
with MII 100M and 10M. For this reason we revert back 10/100M
to use default setting (LMAC).
---
 drivers/net/ethernet/lantiq/switch-api/mac/mac_cfg.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/mac/mac_cfg.c b/drivers/net/ethernet/lantiq/switch-api/mac/mac_cfg.c
index caa087222a3f..9a3f0fe7c6e3 100644
--- a/drivers/net/ethernet/lantiq/switch-api/mac/mac_cfg.c
+++ b/drivers/net/ethernet/lantiq/switch-api/mac/mac_cfg.c
@@ -242,7 +242,6 @@ int mac_set_mii_if(void *pdev, u32 mii_mode)
 		gswss_set_fe_intf(pdev, LMAC_MII);
 	} else if (mii_mode == GSW_PORT_HW_GMII) {
 		gswss_set_1g_intf(pdev, XGMAC_GMII);
-		gswss_set_fe_intf(pdev, XGMAC_GMII);
 		gswss_set_2G5_intf(pdev, XGMAC_GMII);
 	} else if (mii_mode == 5) {	/* For Testing LMAC 1G only */
 		gswss_set_1g_intf(pdev, LMAC_GMII);
@@ -1101,7 +1100,7 @@ int mac_init(void *pdev)
 	/* Default enable flow control Rx and TX */
 	mac_set_flowctrl(pdev, 3);
 #endif
-	/* Configure MII for 100M, 10M, 1G and 2.5G to XGMAC by default
+	/* Configure MII for 1G and 2.5G to XGMAC by default
      * default value is LMAC
      */
 	mac_set_mii_if(pdev, GSW_PORT_HW_GMII);
